en/flight_modes_vtol/return.md

```diff
@@ -0,0 +1,67 @@
+# Return Mode (VTOL)
```

Comment:
@sfuhrer I have VTOL questions for RTL_TYPE=1 (the default)

1. There is a note that says "When this type is set PX4 will reject a mission without a valid landing pattern."
   - I tried this in simulation and it appears to be correct.
   - Doesn't that mean you have to have a landing pattern for VTOL? How does that make sense? A VTOL might want to use a mission with a normal VTOL_LAND - doesn't this preclude it?
2. The original doc defined the behaviour for VTOL RTL_TYPE=1 as:
   > - If a mission landing is defined, fly direct to the mission landing start point and then land.
   > - Otherwise fly directly to the home position, transition to multicopter mode, and land as a multicopter.

   This differs from the general instructions that say "closest of mission landing or any rally point gets run", home gets used if no mission landing or rally point is defined" - my assumption is that the "general instructions" are correct (i.e. rally points are used by VTOL too, in preference to home)?

3. The docs state that:
   >   (for a rallypoint/home location) If not in a mission landing, a VTOL in FW mode will _always_ transition back to MC just before landing (ignoring [NAV_FORCE_VT](../advanced_config/parameter_reference.md#NAV_FORCE_VT)).

   - If we're flying as an FW when return mode is triggered, do we use the FW settings for return altitude - i.e ignore the cone setting? 
   - If we're flying as an MC when return mode is triggered, do we switch to FW to fly back? 
   - If the answer is "no" then do we obey the minimum return altitude of MC, taking account of the cone?
   - When does the transition happen? Does it happen below the descent altitude (i.e. on final landing) or at the descent altitude (i.e. when waiting)?
   
   My guess here is that we use the MC settings in MC mode, FW settings in FW mode. 
   - So if we're in FW when the return mode occurs we fly back at RTL_ALTITUDE (ignoring the cone), spiral down to the descent altitude, start landing, transition to MC, then land. The only difference from FW in return behaviour being no indefinite loiter at the wait point (since we can land) and the transition.
   - If we're in MC mode we'd fly back at the minimum return altitude as an MC, taking account of the cone parameter (which seems to default to 45 deg in SITL), and simply land as normal as an MC. No switching back to FW.
   
   Can you confirm?

Reply:
>There is a note that says "When this type is set PX4 will reject a mission without a valid landing pattern."

This comment is outdated, the param `MIS_TKO_LAND_REQ` governs the requirements now.

> Doesn't that mean you have to have a landing pattern for VTOL? How does that make sense? A VTOL might want to use a mission with a normal VTOL_LAND - doesn't this preclude it?

VTOL_LAND also counts as a landing (now). And if you don't have any landing planned but the type is 1, then it just goes to Home.


> my assumption is that the "general instructions" are correct (i.e. rally points are used by VTOL too, in preference to home)?

Yes rally points are used as well, they are like multiple Home points, and the closest is taken. The parameter description of RTL_TYPE should be correct.


>If we're flying as an FW when return mode is triggered, do we use the FW settings for return altitude - i.e ignore the cone setting?

Don't fully understand what you mean, there is no specific "FW setting for return altitude". Cone is ignored. Vehicle returns as if it would be a fixed-wing vehicle, and then transitions to MC before landing.

I confirm your other points, especially the last two.

Reply:
Thanks. For the landing pattern part, I have taken this to mean that a landing pattern is defined as:

> This consists of a [MAV_CMD_DO_LAND_START](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_LAND_START), one or more position waypoints, and a [MAV_CMD_NAV_LAND](https://mavlink.io/en/messages/common.html#MAV_CMD_NAV_LAND) (or [MAV_CMD_NAV_VTOL_LAND](https://mavlink.io/en/messages/common.html#MAV_CMD_NAV_VTOL_LAND) for a VTOL Vehicle).

So you still need the LAND_START. Presumably an MC that used this would just LAND on the final point.

Reply:
Otherwise, all good I think. Thanks.